src/components/com_tjdashboard/administrator/helpers/dashboard.php

```diff
@@ -25,15 +25,113 @@ class DashboardHelper
 	 */
 	public static function addSubmenu($vName = '')
 	{
-		JHtmlSidebar::addEntry(
-			JText::_('COM_TJDASHBOARD_VIEW_DASHBOARDS'),
-			'index.php?option=com_tjdashboard&view=dashboards',
-			$vName == 'dashboards'
-		);
-		JHtmlSidebar::addEntry(
-			JText::_('COM_TJDASHBOARD_VIEW_WIDGETS'),
-			'index.php?option=com_tjdashboard&view=widgets',
-			$vName == 'widgets'
-		);
+		$layout = JFactory::getApplication()->input->get('layout', '', 'STRING');
+
+		if ($layout != "default")
+		{
+			JHtmlSidebar::addEntry(
+				JText::_('COM_TJDASHBOARD_VIEW_DASHBOARDS'),
+				'index.php?option=com_tjdashboard&view=dashboards',
+				$vName == 'dashboards'
+			);
+			JHtmlSidebar::addEntry(
+				JText::_('COM_TJDASHBOARD_VIEW_WIDGETS'),
+				'index.php?option=com_tjdashboard&view=widgets',
+				$vName == 'widgets'
+			);
+		}
+		else
+		{
+			$client = JFactory::getApplication()->input->get('client', '', 'STRING');
+
+			$full_client = $client;
+
+			// Set ordering.
+			$mainframe = JFactory::getApplication();
+			$full_client = explode('.', $full_client);
+
+			// Eg com_jgive
+			$component = $full_client[0];
+			$eName = str_replace('com_', '', $component);
+			$file = JPath::clean(JPATH_ADMINISTRATOR . '/components/' . $component . '/helpers/' . $eName . '.php');
+
+			if (file_exists($file))
+			{
+				require_once $file;
+
+				$prefix = ucfirst(str_replace('com_', '', $component));
+				$cName = $prefix . 'Helper';
+
+				if (class_exists($cName))
+				{
+					if (is_callable(array($cName, 'addSubmenu')))
+					{
+						$lang = JFactory::getLanguage();
+
+						// Loading language file from the administrator/language directory then
+						// Loading language file from the administrator/components/*extension*/language directory
+						$lang->load($component, JPATH_BASE, null, false, false)
+						|| $lang->load($component, JPath::clean(JPATH_ADMINISTRATOR . '/components/' . $component), null, false, false)
+						|| $lang->load($component, JPATH_BASE, $lang->getDefault(), false, false)
+						|| $lang->load($component, JPath::clean(JPATH_ADMINISTRATOR . '/components/' . $component), $lang->getDefault(), false, false);
+
+						// Call_user_func(array($cName, 'addSubmenu'), 'categories' . (isset($section) ? '.' . $section : ''));
+						call_user_func(array($cName, 'addSubmenu'), $vName . (isset($section) ? '.' . $section : ''));
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Gets a list of the actions that can be performed.
+	 *
+	 * @param   string   $component  The component name.
+	 * @param   string   $section    The access section name.
+	 * @param   integer  $id         The item ID.
+	 *
+	 * @return  JObject
+	 *
+	 * @since   3.2
```

Comment:
@monikaatvowel Check the doc comment. Add the correct since tab. Also, check the return type.